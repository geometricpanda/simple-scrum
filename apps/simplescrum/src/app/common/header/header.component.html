<mat-toolbar class="toolbar" color="primary">

  <button aria-label="Open the Side Menu"
          data-test="open-drawer-button"
          mat-icon-button
          (click)="toggleDrawer()">
    <mat-icon>menu</mat-icon>
  </button>

  <span class="toolbar__title">Simple Scrum</span>

</mat-toolbar>
