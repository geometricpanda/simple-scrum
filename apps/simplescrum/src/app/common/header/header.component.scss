.toolbar {
}

.toolbar__title {
  flex: 1 1 auto;
}
