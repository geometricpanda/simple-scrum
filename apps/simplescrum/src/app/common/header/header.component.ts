import {Component, forwardRef, Inject} from '@angular/core';
import {AppStateService} from '../../app-state.service';
import {BreakpointService} from '../observables/breakpoint.service';

@Component({
  selector: 'app-header',
  templateUrl: './header.component.html',
  styleUrls: ['./header.component.scss'],
})
export class HeaderComponent  {

  // TODO: Figure out why injected dependencies need to be a forwardRef or jest dies
  constructor(@Inject(forwardRef(() => AppStateService)) private appStateService: AppStateService,
              @Inject(forwardRef(() => BreakpointService)) public breakpointService: BreakpointService) {
  }

  async toggleDrawer() {
    await this.appStateService.drawer.toggle();
  }

}
