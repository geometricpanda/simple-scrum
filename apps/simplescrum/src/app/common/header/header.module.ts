import {NgModule} from '@angular/core';
import {CommonModule} from '@angular/common';
import {HeaderComponent} from './header.component';
import {MatButtonModule, MatIconModule, MatToolbarModule} from '@angular/material';
import {BrowserAnimationsModule} from '@angular/platform-browser/animations';


@NgModule({
  declarations: [
    HeaderComponent,
  ],
  imports: [
    CommonModule,
    BrowserAnimationsModule,
    MatToolbarModule,
    MatIconModule,
    MatButtonModule,
  ],
  exports: [
    HeaderComponent,
  ],
})
export class HeaderModule { }
