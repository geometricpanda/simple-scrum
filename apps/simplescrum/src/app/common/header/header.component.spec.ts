import {ComponentFixture, TestBed} from '@angular/core/testing';
import {NoopAnimationsModule} from '@angular/platform-browser/animations';

import {HeaderComponent} from './header.component';
import {HeaderModule} from './header.module';
import {Component} from '@angular/core';
import {By} from '@angular/platform-browser';
import {AppStateService} from '../../app-state.service';

describe('HeaderComponent', () => {

  const appStateServiceMock = {
    drawer: {
      open: jest.fn(),
    },
  };

  const selectors = {
    openDrawerButton: By.css('[data-test="open-drawer-button"]'),
  };

  @Component({
    selector: 'app-header-spec',
    template: `
      <app-header></app-header>
    `,
  })
  class HeaderComponentSpec {
  }

  let component: HeaderComponentSpec;
  let fixture: ComponentFixture<HeaderComponentSpec>;
  let appStateService: AppStateService;

  beforeEach(async () => {
    TestBed.configureTestingModule({
      declarations: [HeaderComponentSpec],
      imports: [HeaderModule, NoopAnimationsModule],
    });
    await TestBed.compileComponents();
  });

  beforeEach(() => {
    fixture = TestBed.createComponent(HeaderComponentSpec);
    component = fixture.componentInstance;
    appStateService = TestBed.inject(AppStateService);
    fixture.detectChanges();
  });

  it('should match the snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  describe('Open Drawer button', () => {
    it('should call the toggle method of the drawer on click', () => {
      appStateService.drawer = {toggle: jest.fn()} as any;

      const $elem = fixture.debugElement.query(selectors.openDrawerButton);
      const $event = new MouseEvent('click');
      $elem.triggerEventHandler('click', $event);

      expect(appStateService.drawer.toggle).toHaveBeenCalled();
    });
  });

});
