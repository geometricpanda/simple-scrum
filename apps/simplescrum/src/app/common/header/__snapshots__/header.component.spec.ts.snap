// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderComponent should match the snapshot 1`] = `
<app-header-spec>
  <app-header>
    <mat-toolbar
      class="toolbar mat-toolbar mat-primary mat-toolbar-single-row"
      color="primary"
    >
      
      <button
        aria-label="Open the Side Menu"
        class="mat-icon-button mat-button-base _mat-animation-noopable"
        data-test="open-drawer-button"
        mat-icon-button=""
      >
        <span
          class="mat-button-wrapper"
        >
          <mat-icon
            aria-hidden="true"
            class="mat-icon notranslate material-icons mat-icon-no-color"
            role="img"
          >
            menu
          </mat-icon>
        </span>
        <div
          class="mat-button-ripple mat-ripple mat-button-ripple-round"
          matripple=""
        />
        <div
          class="mat-button-focus-overlay"
        />
      </button>
      <span
        class="toolbar__title"
      >
        Simple Scrum
      </span>
    </mat-toolbar>
  </app-header>
</app-header-spec>
`;
